packages/realm/src/binding/PolyfilledBigInt.ts

```diff
@@ -0,0 +1,48 @@
+////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2024 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+import { Long } from "bson";
+
+import type { binding } from "./wrapper.generated";
+import { assert } from "../assert";
+
+export const PolyfilledBigInt: typeof binding.Int64 = Object.freeze({
+  add(a: binding.Int64, b: binding.Int64) {
+    assert.instanceOf(a, Long);
+    assert.instanceOf(b, Long);
+    return a.add(b) as unknown as binding.Int64;
+  },
+  equals(a: binding.Int64, b: binding.Int64) {
+    assert.instanceOf(a, Long);
+    assert.instanceOf(b, Long);
+    return a.equals(b);
+  },
```

Comment:
If updating the previous `equals()` to allow numbers and strings, this would likely need to be updated as well, perhaps something like:
(BSON's Long `equals()` also takes numbers and strings)
```suggestion
  equals(a: binding.Int64, b: binding.Int64 | number | string) {
    assert.instanceOf(a, Long);
    assert(
      a instanceof Long || typeof a === "number" || typeof a === "string",
      "Expected a 'BSON.Long', or number, or string.",
    );
    return a.equals(b);
  },
```